server.js

```diff
@@ -0,0 +1,237 @@
+            }
+            req.on('end', () => {
+              res.writeHead(200, {
+                'content-type': `text/${type}`,
+                'content-length': dataStr.length,
+                'Connection': 'keep-alive'
+              });
+            });
+            res.write(dataStr);
+            res.end(); 
+          });
+       });
+      } else if (method === 'POST') {
+        let body = '';
+        req.on('data', (chunk) => {
+          body += chunk.toString();
+        });
+        req.on('end', () => {
+          let bodyParse = querystring.parse(body);
+          let html = `<!DOCTYPE html>
+          <html lang="en">
+          <head>
+            <meta charset="UTF-8">
+            <title>The Elements - ${bodyParse.elementName}</title>
+            <link rel="stylesheet" href="/styles.css">
+          </head>
+          <body>
+            <h1>${bodyParse.elementName}</h1>
+            <h2>${bodyParse.elementSymbol}</h2>
+            <h3>Atomic number ${bodyParse.elementAtomicNumber}</h3>
+            <p>${bodyParse.elementDescription}</p>
+            <p><a href="/">back</a></p>
+          </body>
+          </html>`;
+          let htmlFileName = bodyParse.elementName.charAt(0).toLowerCase() + bodyParse.elementName.slice(1);
+          fs.writeFile(`./public/${htmlFileName}.html`, html, (err, data) => {
+            if (err){return console.log(err);}
+            fs.readdir('./public', (err, dir) => {
+              if (err){return console.log(err);}
+              let dirHtml = dir.filter(url => url.includes('.html') && !url.includes('index') && !url.includes('notFound'));
+              let orderedList = ``;
+              for (let i in dirHtml){
+                orderedList += `
+                <li>
+                  <a href="/${dirHtml[i]}">${dirHtml[i].charAt(0).toUpperCase() + dirHtml[i].slice(1, -5)}</a>
+                </li>`
+              }
+              fs.readFile('./public/index.html', (err, data) => {
+                if (err) {return console.log('could not write the file');}
+                let index = `<!DOCTYPE html>
+                <html lang="en">
+                <head>
+                  <meta charset="UTF-8">
+                  <title>The Elements</title>
+                  <link rel="stylesheet" href="/styles.css">
+                </head>
+                <body>
+                  <h1>The Elements</h1>
+                  <h2>These are all the known elements.</h2>
+                  <h3>These are ${dirHtml.length }</h3>
+                  <ol>
+                   ${orderedList}
+                  </ol>
+                </body>
+                </html>`;
+                fs.writeFile('./public/index.html', index, (err) => {
+                  if (err){return console.log(err);}
+                });
+                req.on('end', () => {
+                  res.writeHead(200, {
+                    'content-type': 'application/json'
+                  });
+                });
```

Comment:
This req.on('end') doesn't make sense since I'm not sure when it'll fire, also it's not needed. But the contents within it are still needed.